File: Julia/SVMClassifier/Kernel.jl

function k_function(x, y, param)
  kernel_type = param.kernel_type
  #RBF
  if kernel_type == 3

    sum = 0
    for i=1:length(x)
      sum += (x[i]-y[i])*(x[i]-y[i])
    end

    return exp(-param.gamma*sum)

  else
    error("ERROR: This type of kernel function is not supported yet")
  end
  return 0
end

File: Python/SVMClassifier/Kernel.py
from math import exp, tanh
from enum import Enum
__author__ = 'Krystian'


class KernelTypes(Enum):
    LINEAR = 0
    POLY = 1
    RBF = 2
    SIGMOID = 3
    PRECOMPUTED = 4


def k_function(x, y, param):
    kernel_type = KernelTypes(param.kernel_type)
    if kernel_type == KernelTypes.RBF:
        sum = 0
        for i in range(0, y.__len__()):
            sum += (x[i]-y[i])*(x[i]-y[i])

        return exp(-param.gamma*sum)

    else:
        print ("ERROR: This type of kernel function is not supported yet")
        return 0
